kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/SchemaManager.java

```diff
@@ -495,6 +495,15 @@ TableInfo constructTableInfo(TableId table, com.google.cloud.bigquery.Schema big
   }
 
   private com.google.cloud.bigquery.Schema getBigQuerySchema(Schema kafkaKeySchema, Schema kafkaValueSchema) {
+    // When converting schema from the last record of a batch, if the record is a tombstone with
+    // null value schema, only allow it to pass if schema unionization is enabled.
+    if (kafkaValueSchema == null) {
+      if (!allowSchemaUnionization) {
+        throw new BigQueryConnectException(
+            "Cannot convert schema for record with value schema.");
```

Comment:
```suggestion
            "Cannot create/update BigQuery table for record with no value schema. "
            + "If delete mode is enabled, it may be necessary to enable schema unionization to handle this case.");
```